### PR TITLE
Deprecate and provide replacement implementation for several gcloud m…

### LINKED ISSUE #44: Update CloudSdkAppEngineModuleService to use non-deprecated gcloud commands
Several of these `gcloud preview app modules _` commands have been deprecated and migrated to `gcloud preview app services`

### PR DESCRIPTION
**Pull request by etanshaul**
…odule services

fixes #44 

Deprecates and replaces most services under `gcloud preview app modules`

Still pending:

getLogs()
setManagedBy()

### CODE REVIEW

**Comment by meltsufin** [comment]
I think I would rather just remove the methods that are deprecated. No one is using this part of the library yet anyway.

**Comment by etanshaul** [comment]
thats fine. I was leaving them in for convenience during review. I can remove them before merging

**Comment by meltsufin** [comment]
OK, sounds good.

**Comment by meltsufin** [comment]
It's looking good... Would you mind adding some tests though? I know the modules code didn't have any, but we're trying to improve. :smile:

**Comment by etanshaul** [comment]
Absolutely. I was actually working one for Args as we speak.

**Comment by meltsufin** [comment]
Awesome!!! :+1:

File: src/main/java/com/google/cloud/tools/app/api/version/AppEngineVersion.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ */
+
+package com.google.cloud.tools.app.api.version;
+
+import com.google.cloud.tools.app.api.AppEngineException;
+
+/**
+ * This set of commands can be used to manage existing App Engine versions.
+ */
+public interface AppEngineVersion {
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/api/version/AppEngineVersion.java]: Version**s** ?

File: src/main/java/com/google/cloud/tools/app/api/version/ListConfiguration.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.cloud.tools.app.api.version;
+
+/**
+ * Configuration for {@link AppEngineVersion#list(ListConfiguration)}.
+ */
+public interface ListConfiguration {
```
Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/api/version/ListConfiguration.java]: I wonder if VersionsListConfiguration is more obvious here.

File: src/test/java/com/google/cloud/tools/app/impl/cloudsdk/util/ArgsTest.java [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+import java.util.Collections;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import static org.junit.Assert.assertEquals;
+
+/**
+ * Unit tests for {@link Args}
+ */
+public class ArgsTest {
```
Comment (loosebazooka) [inline src/test/java/com/google/cloud/tools/app/impl/cloudsdk/util/ArgsTest.java]: Oooh, I should add the other tests for this... good call : #52

File: src/main/java/com/google/cloud/tools/app/api/version/AppEngineVersion.java (hunk 1 above)
Comment (etanshaul) [inline src/main/java/com/google/cloud/tools/app/api/version/AppEngineVersion.java]: Agreed. I'll also update AppEngineService**s**

File: src/main/java/com/google/cloud/tools/app/api/version/ListConfiguration.java (hunk 2 above)
Comment (etanshaul) [inline src/main/java/com/google/cloud/tools/app/api/version/ListConfiguration.java]: Yea. I'll also update its default implementation to be DefaultVersionsListConfiguration.

**Comment by etanshaul** [comment]
Address naming comments and added unit tests.

**Comment by etanshaul** [comment]
@loosebazooka - since we updated VersionsListConfiguration to be plural, I wonder if we should also update Version**s**SelectionConfiguration to be plural (and all related config classes). I think singular is fine for all them, but I'm open to either. what do you think?

**Comment by meltsufin** [comment]
+1 for using plurals everywhere

**Comment by loosebazooka** [comment]
Sure plurals everywhere

**Comment by etanshaul** [comment]
I went ahead and deprecated getLogs() and setManagedBy()

If you look carefully at the new services, not all of them are a direct 1-to-1 mapping from the deprecated ones. For this reason, I think it may be good to leave the deprecated services in for now. Let me know which you prefer.

**Comment by loosebazooka** [comment]
Take the deprecated ones out. We aren't consuming them.

**Comment by etanshaul** [comment]
done - everything from gcloud preview app modules is gone

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineInstances.java [hunk 4]
```diff
@@ -0,0 +1,91 @@
...
+    setDebug(configuration, false);
+  }
+
+  private void setDebug(InstancesSelectionConfiguration configuration, boolean enable) {
+    Preconditions.checkNotNull(configuration);
+    Preconditions.checkNotNull(configuration.getVersion());
+    Preconditions.checkNotNull(sdk);
+
+    List<String> arguments = new ArrayList<>();
+    arguments.add("instances");
```
Comment (meltsufin) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineInstances.java]: Can you use the Args utility in all these places, even though Modules code wasn't using it?

Comment (etanshaul) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineInstances.java]: Yes. I missed a few it looks like. I'll update it.

**Comment by etanshaul** [comment]
@meltsufin ok I think I got them.

**Comment by meltsufin** [comment]
Yep, looks good. Once more thing: should we rename the packages (instance, log, service, version) to plurals?

**Comment by etanshaul** [comment]
Yes, done.

**Comment by meltsufin** [comment]
LGTM :ship: :rocket:

**Comment by etanshaul** [comment]
@loosebazooka if all your comments have been addressed I'll merge this in

**Comment by loosebazooka** [comment]
:ship:

**Comment by loosebazooka** [comment]
can you make sure the sqash+merge commit is meaningful?

**Pull request merged**